### PR TITLE
fix: add newline in assign before `?:` operator

### LINKED ISSUE #329: Odd linebreaks for ?: operator (not ternary)
PHP: 7.0
Node: 9
prettier-php: 26c6262
prettier: <code>[d676188](https://github.com/prettier/prettier/commit/d676188dbb0c99de57bdffda96ab0cccb1abfa1d)</code>

### Input
```php
<?php
class Horse {
	protected function get_category_color( $term_id ) {
		$category_color = get_field( Category_Meta::COLOR, 'category_' . $term_id ) ?: 'gold';
		return $category_color;
	}
}
```

### Output
```php
class Horse
{
	protected function get_category_color( $term_id )
	{
		$category_color = get_field(
			Category_Meta::COLOR,
			'category_' . $term_id
		)
			?: 'gold';
		return $category_color;
	}
}
```

### Expected Output
Uh… Something like this?

```php
class Horse
{
	protected function get_category_color( $term_id )
	{
		$category_color =
			get_field(Category_Meta::COLOR, 'category_' . $term_id) ?: 'gold';
		return $category_color;
	}
}
```

--- 

If I change `?:` for `??`, which is a similar operator, I get what I put under "Expected Output".

I don't think the current output is intended, but I'm not attached to my "expected output". I only put it there because prettier-php will produce it for a similar input, so it seems consistent. (and nicer than the current output.)

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #329

### CODE REVIEW

File: tests/class/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -313,8 +313,8 @@ abstract class ReallyReallyReallyLongClassName
     {
         parent::__construct($test_int ?: 1);
         $this->other = $test_string;
-        $this->current_version = $current_version
-            ?: new Content_Version_Model();
+        $this->current_version =
+            $current_version ?: new Content_Version_Model();
```
Comment (evilebottnawi) [inline tests/class/__snapshots__/jsfmt.spec.js.snap]: From prettier with same operator from typescript`:
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASGALAlgZwHRgCuATsQjAPoBucxOW0ABALyOpGnnW33QA6URkOEjRQgPzjGUOAHdGAYWjxYFAGo8GUCgFkIAQ0NHjJkwBM4AGwAUASgDcAkABoQEAA4wtOZKH2kIWQAFfwQfFH0qCCwzFxB9HBhkADN9Sxw4VwAjYn0wAGs4GABldzysKABzZBhiQkyQMwgwFLSM1wqM4hgg3MqAW31W9IaAKxwADwAhXIKi4v1+uAAZCrhh9rdCGHdtgCYNhrK6WmQQLP0sgE9LaDj3YgqYAHUYzGQADgAGVweIDOeuXcZwecC6NDiZAAjoQsGRevoBkMkKkRq4Mv0sDU6g16FVLHAAIqECDwQ6uGCXV5md5IPYU3JYSwVSpKfqDM5QaDrVyEDIAFUu4VRGQAvqKgA

File: tests/if/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -69,16 +69,6 @@ if (($this || $that) && $theOtherThing < $someReallyReallyReallyLongBooleanVaria
   return true;
 }
 
-return true ? 1 : 2;
```
Comment (evilebottnawi) [inline tests/if/__snapshots__/jsfmt.spec.js.snap]: Refactor tests, move all retif in own directory

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -456,9 +456,9 @@ do {
 } while (true);
 
 $test = foo ?: bar(function () {
-        $foo = "bar";
-        // test
-    });
+            $foo = "bar";
+            // test
+        });
```
Comment (evilebottnawi) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: WIP

**Comment by evilebottnawi** [comment]
Bug in parser https://github.com/glayzzle/php-parser/issues/145

File: tests/retif/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+$category_color =
+    get_field(Category_Meta::COLOR, 'category_' . $term_id) ?: 'gold';
+
+$var = $someOtherReallyReallyLongVariable
+? $someOtherReallyReallyLongVariable
+? $someOtherReallyReallyLongVariable
+: $someOtherReallyReallyLongVariable
+: $someOtherReallyReallyLongVariable
+    ? $someOtherReallyReallyLongVariable
+    : $someOtherReallyReallyLongVariable;
```
Comment (evilebottnawi) [inline tests/retif/__snapshots__/jsfmt.spec.js.snap]: https://github.com/glayzzle/php-parser/issues/145

File: tests/retif/__snapshots__/jsfmt.spec.js.snap [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+$test = $foo ?: bar(function () {
+        $foo = "bar";
+        // test
+    });
```
Comment (evilebottnawi) [inline tests/retif/__snapshots__/jsfmt.spec.js.snap]: Looks ugly, let's fix it in this PR and review welcome :+1:

**Comment by evilebottnawi** [comment]
//cc @mgrip @czosel ready fore review :+1:

File: src/printer.js [hunk 6]
```diff
@@ -1035,8 +1035,8 @@ function printStatement(path, options, print) {
   switch (node.kind) {
     case "assign": {
       const canBreak =
-        node.right.kind === "bin" ||
-        ["number", "string"].indexOf(node.right.kind) > -1 ||
+        ["bin", "number", "string"].indexOf(node.right.kind) > -1 ||
```
Comment (mgrip) [inline src/printer.js]: `["bin", "number", "string"].includes(node.right.kind)`

Comment (evilebottnawi) [reply src/printer.js]: @mgrip yes :+1:

**Pull request merged**